src/plots/gl2d/scene2d.js

```diff
@@ -286,10 +292,11 @@ proto.updateFx = function(options) {
 
     fullLayout.dragmode = options.dragmode;
     fullLayout.hovermode = options.hovermode;
-};
 
-var relayoutCallback = function(scene) {
+    this.graphDiv._fullLayout = fullLayout;
```

Comment:
I believe the problem is in the [`doModebar`](https://github.com/plotly/plotly.js/blob/5e4bb1a49f93e59e682a142099923dad3a21f861/src/plot_api/subroutines.js#L293-L319) relayout shortcut. 

I'll double-check why the `_fullLayout` gets lost. Thanks for the headsup!